### ISSUE TITLE
Missing Origin Validation during npm install

### ISSUE THREAD

**Issue by tpw1314** [webpack/webpack-dev-server#1566]
* Operating System: Windows 10
* Node Version: v11.1.0
* NPM Version: 6.4.1
* webpack Version: 3.12.0
* webpack-dev-server Version: 3.1.10

- [x] This is a **bug**
- [ ] This is a **modification** request

### Code
N/A

### Expected Behavior
npm install finishes without error

### Actual Behavior
```npm install finishes with the following error ::
npm WARN deprecated browserslist@2.11.3: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.
npm WARN deprecated browserslist@1.7.7: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.

> node-sass@4.10.0 install C:\xampp\htdocs\laravue\node_modules\node-sass
> node scripts/install.js

Cached binary found at C:\Users\User\AppData\Roaming\npm-cache\node-sass\4.10.0\win32-x64-67_binding.node

> uglifyjs-webpack-plugin@0.4.6 postinstall C:\xampp\htdocs\laravue\node_modules\webpack\node_modules\uglifyjs-webpack-plugin
> node lib/post_install.js


> node-sass@4.10.0 postinstall C:\xampp\htdocs\laravue\node_modules\node-sass
> node scripts/build.js

Binary found at C:\xampp\htdocs\laravue\node_modules\node-sass\vendor\win32-x64-67\binding.node
Testing binary
Binary is fine
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN img-loader@3.0.1 requires a peer of imagemin@^5.0.0 || ^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN ajv-keywords@3.2.0 requires a peer of ajv@^6.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.4 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.4: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

added 1189 packages from 698 contributors and audited 11710 packages in 196.717s
found 1 high severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
```

### For Bugs; How can we reproduce the behavior?
On Windows environment, launch cmd and run the following commands ::
```
composer create-project --prefer-dist laravel/laravel laravel
```
And then run 
```
npm install
```

**Comment by tpw1314** [comment]
I remember the first time I installed laravel via composer + npm install, there was not a problem at all. However, I had to re-install a fresh laravel project so I ended up just 'Right click' and delete the project folder.
After few busy weeks I now decided to have fresh laravel project and then run ```npm install``` and boom.

**Comment by evilebottnawi** [comment]
Please this create issue in `laravel-mix`. you use old `webpack-dev-server` version in `laravel-mix`

**Comment by tanx** [comment]
Any chance the security fix will get backported to `webpack-dev-server@2.x.x`? We're still on `react-scripts@1.x.x` yet which relies on this version range. Thanks for your consideration.

````
=== npm audit security report ===                        
                                                                                
# Run  npm install react-scripts@2.1.1  to resolve 1 vulnerability
SEMVER WARNING: Recommended action is a potentially breaking change
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Missing Origin Validation                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ react-scripts                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ react-scripts > webpack-dev-server                           │
├───────────────┼─���────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/725                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
````

**Comment by evilebottnawi** [comment]
@tanx `webpack-dev-server@2` is deprecated and not updated anymore due a lot of bugs and security problem, sorry

**Comment by tpw1314** [comment]
For all of you, this issue appears to be solved by this [#1815](https://github.com/JeffreyWay/laravel-mix/pull/1815)

**Comment by tanx** [comment]
Not in `react-scripts@1.x.x` unfortunately though.

**Comment by tpw1314** [comment]
Please refer to [this issue](https://github.com/JeffreyWay/laravel-mix/issues/1809). Apparently it is not fixed yet.

**Comment by connecteev** [comment]
Seeing the same. This issue hasn't been fixed yet.

**Comment by Download** [comment]
Same here

**Comment by proliuk** [comment]
=== npm audit security report ===


                                 Manual Review
             Some vulnerabilities require your attention to resolve

          Visit https://go.npm.me/audit-guide for additional guidance
```

 High            Missing Origin Validation

  Package         webpack-dev-server

  Patched in      >=3.1.6

  Dependency of   laravel-mix [dev]

  Path            laravel-mix > webpack-dev-server

  More info       https://nodesecurity.io/advisories/725

found 1 high severity vulnerability in 15147 scanned packages
  1 vulnerability requires manual review. See the full report for details.
```

Already, what we can do with this vulnerabilities? Do it works in a local server?

**Comment by DawgOnKing** [comment]
Still not resolved as of this evening. Fresh laravel install on Amazon AMI still shows report. Working locally but I haven't tried to deploy.

**Comment by SharakPL** [comment]
`npm install laravel-mix@beta` fixes the problem

**Comment by enbermudas** [comment]
@SharakPL options actually fixes the problem. Thanks a lot!

**Comment by Jvalal** [comment]
this did not fix it for me. 

![image](https://user-images.githubusercontent.com/3803053/49777129-f2f09a00-fcb3-11e8-89cc-f71c42ba16d3.png)

**Comment by LaravDev** [comment]
> `npm install laravel-mix@beta` fixes the problem

i did this and worked, but i have some kind of issue with babel-core now, guess it's a separated issue

**Comment by tpw1314** [comment]
Anyone manage to run npm install laravel-mix without hitting any error even after a month ?

**Comment by deepeshmadkar** [comment]
yes i am also getting the same issues.

**Comment by klymenkoo** [comment]
I can confirm. I'm using create-react-app.

**Comment by ealexhaywood** [comment]
As am I, latest CRA is using `webpack-dev-server@3.1.9` so I notified them in the existing issue that was opened https://github.com/facebook/create-react-app/issues/5777#issuecomment-450684848

**Comment by amitozdeol** [comment]
Issue still persist. Using fresh copy of create-react-app on Mac OS High sierra

**Comment by dvvdev** [comment]
Issue is occuring with vue/cli-service

**Comment by tetrohed** [comment]
having the same issue with creat-react-app! can somebody clarify why this is closed?! The react issue links to here and is closed as well

**Comment by ealexhaywood** [comment]
Everyone who is using create react app should post here https://github.com/facebook/create-react-app/issues/5777.  Webpack has already said they are not back porting fixes.  Tell create-react-app to update their webpack dev server version.

**Comment by millenjo** [comment]
I'm getting this directly when just installing webpack-dev-server. I create a new folder, run `npm init` then `npm install webpack-dev-server --save` and i get:

```
                       === npm audit security report ===

┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Missing Origin Validation                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.1.11                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/725                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 1 high severity vulnerability in 3441 scanned packages
  1 vulnerability requires manual review. See the full report for details.
```

when i run `npm audit`

**Comment by palkan** [comment]
> is NPM maybe interpreting 3.1.1* as lower that 3.1.6?

just tried 3.1.9 – the same(

**Comment by falsy** [comment]
If you run 'npm install' on Node v8.12.0, the following error appears. (npm v6.4.1)

```                       === npm audit security report ===

┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ High          │ Missing Origin Validation                                    │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=3.1.11                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ webpack-dev-server [dev]                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ webpack-dev-server                                           │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://nodesecurity.io/advisories/725                       │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 1 high severity vulnerability in 10268 scanned packages
  1 vulnerability requires manual review. See the full report for details.```

'npm install' on Node v6.14.4 does not output an error. (npm v3.10.10)

**Comment by fergyfresh** [comment]
@falsy. we are seeing the same thing. I've retried it with `3.1.11` through `3.1.14` and no dice.

**Comment by simPod** [comment]
Those versions are listed as affected

![image](https://user-images.githubusercontent.com/327717/50597802-ecb39f00-0ea8-11e9-9882-daef67fc1bcd.png)

**Comment by evalineBai** [comment]
This was due to a typo in the vulnerability database apparently: https://npm.community/t/advisory-725-inconsistently-marks-affected-versions/4333/3

**Comment by simPod** [comment]
On the 1st January I contacted NPM. Should be just fine now

![image](https://user-images.githubusercontent.com/327717/50601985-41114b80-0eb6-11e9-8c6b-8cfb0806f08c.png)
![image](https://user-images.githubusercontent.com/327717/50601990-4a021d00-0eb6-11e9-9e26-e14244903cac.png)

**Comment by falsy** [comment]
> @falsy. we are seeing the same thing. I've retried it with `3.1.11` through `3.1.14` and no dice.

@fergyfresh Hi, fergyfresh.

First of all, I am sorry but I am not good at English. (I didn't understand the meaning of 'dice'.)
I am currently using v3.1.14. Most webpack-dev-server v3.1.x output an error on node v8.12.0. However, we have used the NVM to change to Node v6.14.4 and have verified that it does not output subsequent installed result errors.

I don't think this is the solution. 
If you need more information from me, let me know.
Thank you.

**Comment by fergyfresh** [comment]
@falsy it just means it did not work. Still have the high vulnerability security warning.

**Comment by falsy** [comment]
@fergyfresh I tested it on my PC at home and found it installed on node v8.12.0 without errors. 
Sorry, I think the cause of the problem is something other than the node version.

**Comment by simPod** [comment]
@falsy Have you read my posts? I think it explains it all.

**Comment by SharakPL** [comment]
@simPod I can confirm that it's been fixed. `npm audit` is clean now.

**Comment by falsy** [comment]
@simPod Oh, it's fixed. thank you.

**Comment by falsy** [comment]
@simPod Oh, now the error is not output. Thank you.

**Comment by quang5566** [comment]
I'm also getting the same issues. Althought I was update `webpack-dev-server@3.1.14 `. But I can't run `npm run dev`. `TypeError: Cannot destructure property `compile` of 'undefined' or 'null'.`

**Comment by bandisudhir** [comment]
I'm also still seeing the issue. If anyone found a solution please help.

**Comment by jduser36** [comment]
I´m also getting this isusue on a mac. I am new to react.js, so is there a way to solve it or to bypass it?

**Comment by jduser36** [comment]
Yes it is, and i have the version 3.5.0 installed.

**Comment by bandisudhir** [comment]
> I'm also still seeing the issue. If anyone found a solution please help.

After updating the webpack-dev-server@3.1.14. The problem is resolved though it is still showing 1 high vulnerability I was able to install required packages. If your problem is not resolved delete node_modules folder and install again. Hoping your problem will be resolved.

**Comment by bandisudhir** [comment]
> Yes it is, and i have the version 3.5.0 installed.

Please check the version properly the highest version of webpack-dev-server available is 3.1.14 when I'm writing this comment.
